Sip import fix

draperjames opened the pull request:
@zbarge try testing out this fork when you get the chance and let me know how it work for you. I haven't had any problems.

Commit:
A qtpy based compat.py solution and trash removal

changed compat.py to rely on the qtpy package and deleted some of the
files listed in #7

Commit:
Forgot compat

Commit:
requirements changes

Added qtpy to reqirements, cleaned up compat.py

zbarge commented:
OK about to test now.

draperjames commented:
If anything breaks in your env would you mind listing it here?

zbarge commented:
It works OK for me - i get 9 failed tests though and "cannot import sip" still

zbarge commented:
test accept/reject for CSVDialogs:

qapp = <PySide.QtGui.QApplication object at 0x00000000069AAE08>
request = <SubRequest 'qtbot' for <Function 'test_accept_reject'>>

    @pytest.yield_fixture
    def qtbot(qapp, request):
        """
        Fixture used to create a QtBot instance for using during testing.
    
        Make sure to call addWidget for each top-level widget you create to ensure
        that they are properly closed after the test ends.
        """
        result = QtBot(qapp)
        no_capture = request.node.get_marker('qt_no_exception_capture') or \
                     request.config.getini('qt_no_exception_capture')
        if no_capture:
            yield result  # pragma: no cover
        else:
            with capture_exceptions() as exceptions:
                yield result
            if exceptions:
>               pytest.fail(format_captured_exceptions(exceptions))
E               Failed: Qt exceptions in virtual methods:
E               ________________________________________________________________________________
E                 File "C:\Users\Zeke\Google Drive\dev\python\qtpandas\qtpandas\views\CSVDialogs.py", line 657, in rejected
E                   self.exported.emit(False)
E               
E               TypeError: Value types used on meta functions (including signals) need to be registered on meta type: QBool
E               ________________________________________________________________________________

zbarge commented:
same test accept reject

Traceback (most recent call last):
  File "C:\Users\Zeke\Google Drive\dev\python\qtpandas\qtpandas\views\CSVDialogs.py", line 657, in rejected
    self.exported.emit(False)
TypeError: Value types used on meta functions (including signals) need to be registered on meta type: QBool
F
self = <tests.test_CSVDialogs.TestCSVExportWidget object at 0x00000000067F15F8>
qtbot = <pytestqt.plugin.QtBot object at 0x00000000067F1198>

    def test_accept_reject(self, qtbot):
        csvwidget = CSVExportDialog()
        qtbot.addWidget(csvwidget)
        csvwidget.show()
    
        buttons = csvwidget.findChildren(QtGui.QPushButton)
        for button in buttons:
            qtbot.mouseClick(button, QtCore.Qt.LeftButton)
            if button.text() == 'Export Data':
                assert csvwidget.isVisible() == True
            else:
>               assert csvwidget.isVisible() == False
E               assert True == False
E                +  where True = <built-in method isVisible of CSVExportDialog object at 0x0000000006807048>()
E                +    where <built-in method isVisible of CSVExportDialog object at 0x0000000006807048> = <qtpandas.views.CSVDialogs.CSVExportDialog object at 0x0000000006807048>.isVisible

test_CSVDialogs.py:269: AssertionError

zbarge commented:
F
def test_TimestampFormat():
        model = DataFrameModel()
        assert model.timestampFormat == Qt.ISODate
        newFormat = "yy-MM-dd hh:mm"
        model.timestampFormat = newFormat
        assert model.timestampFormat == newFormat
    
        with pytest.raises(TypeError) as excinfo:
>           model.timestampFormat = "yy-MM-dd hh:mm"
E           Failed: DID NOT RAISE <class 'TypeError'>

test_DataFrameModel.py:66: Failed

Commit:
another attempt at a qtpy solution

HOTTIP: delete one thing at a time
Don't be like me if you need to remove somethings do it methodically.

Commit:
Closing in on a solution

Moved the contents of test/main.py to test/test_DataFrameModelManager.py
that way I can actually run the tests after a fresh install. Commented
out some requirements in setup.py.

draperjames commented:
Okay try to do a fresh install of this last commit. I managed to get all of the tests to at least open at last count I have 5 failed, 147 passed, 2 errors.

draperjames commented on MANIFEST.in:
```diff
@@ -1,4 +1,3 @@
 recursive-include qtpandas *.py
 recursive-include qtpandas/_lib *
-recursive-include qtpandas/i18n *.qm
```
Deleted this line b/c I deleted the corresponding directory.

draperjames commented on qtpandas/compat.py:
```diff
@@ -1,34 +1,17 @@
-    from PySide.QtCore import Slot, Signal
-
-
-QtCore = QtCore_
```
What's the difference b/t `_QtCore` and `QtCore_`?

zbarge replied:
I dont think there would be any difference just looks like the variable name the author chose.

draperjames commented on qtpandas/encoding.py:
```diff
@@ -8,41 +8,3 @@
     lib = os.path.join(BASEDIR, '_lib', 'magic')
```
All of this should probably be deleted.

zbarge commented:
I just fetched the latest and im testing it now  and failing 9/154.

I think we should change the signals to regular python types, right now there are strings like the following:

Line 501 qtpandas/qtpandas/views/CSVDialogs.py
        exported = Signal('QBool')

I want to change 'QBool' to bool on all the signals. Do you know any reason why I shouldn't do that?


Error in test: 

E               ________________________________________________________________________________
E                 File "C:\Users\Zeke\Google Drive\dev\python\qtpandas\qtpandas\views\CSVDialogs.py", line 657, in rejected
E                   self.exported.emit(False)
E               
E               TypeError: Value types used on meta functions (including signals) need to be registered on meta type: QBool

draperjames commented:
The only reason I can think is that Qbool might be some kind of C++ implementation of a bool. IDK go ahead and try it. I'll poke around and see what I can find about Qbool.

Commit:
Mixed results changing from Qbool to bool

Commit:
Changing 'bool' to bool and a few PEP8 tweaks

I realized that I changed 'QBool' to 'bool' when it probably should have
been just bool.

zbarge commented on qtpandas/views/CSVDialogs.py:
```diff
@@ -120,7 +120,7 @@ def _initUI(self):
         self.setLayout(layout)
 
-    @Slot('QBool')
+    @Slot('bool')
```
I meant bool as in the python type rather than a string that says 'bool' - sorry lol.. @draperjames

draperjames commented:
Yeah it has worked pretty well for me. I also went back to the original pandas-qt repo and installed it in a python27 env, ran pytest on tests, it hit error pretty early on in in the cycle and exited pytest. So now I'm thinking that I either cocked-up the install or we have actually made some real improvements!

Commit:
Fixed error in excpthook & removed some tests

Commit:
Commented out from future stuff

Commit:
attempting to run pytest from travis

Commit:
giving up on pytest through travis for now

Commit:
operator fix and commenting out icons_rc useage
